From a4e965598ffd92dd025b327ebaa4bd19afb6a294 Mon Sep 17 00:00:00 2001
From: Jan Cholasta <jcholast@redhat.com>
Date: Wed, 6 Mar 2013 10:07:13 +0100
Subject: [PATCH 105/116] Remove disabled entries from sudoers compat tree.

The removal is triggered by generating an invalid RDN when ipaEnabledFlag of
the original entry is FALSE.

https://fedorahosted.org/freeipa/ticket/3437
---
 install/share/schema_compat.uldif       | 2 +-
 install/updates/10-schema_compat.update | 2 ++
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/install/share/schema_compat.uldif b/install/share/schema_compat.uldif
index deca1bb416c352ee6cbca0a67576706a317d8b2e..33c3a3654ea9def68e40800eca081d81aaa49552 100644
--- a/install/share/schema_compat.uldif
+++ b/install/share/schema_compat.uldif
@@ -69,7 +69,7 @@ add:cn: sudoers
 add:schema-compat-container-group: 'ou=SUDOers, $SUFFIX'
 add:schema-compat-search-base: 'cn=sudorules, cn=sudo, $SUFFIX'
 add:schema-compat-search-filter: (&(objectclass=ipaSudoRule)(!(compatVisible=FALSE))(!(ipaEnabledFlag=FALSE)))
-add:schema-compat-entry-rdn: cn=%{cn}
+add:schema-compat-entry-rdn: %ifeq("ipaEnabledFlag", "FALSE", "DISABLED", "cn=%{cn}")
 add:schema-compat-entry-attribute: objectclass=sudoRole
 add:schema-compat-entry-attribute: 'sudoUser=%ifeq("userCategory","all","ALL","%{externalUser}")'
 add:schema-compat-entry-attribute: 'sudoUser=%ifeq("userCategory","all","ALL","%deref_f(\"memberUser\",\"(objectclass=posixAccount)\",\"uid\")")'
diff --git a/install/updates/10-schema_compat.update b/install/updates/10-schema_compat.update
index 9835bb8ce2007b34085f1f9e445cf2ce07f2f01c..e65e67afc40f10eee342afc12e2132d36e9c2a1a 100644
--- a/install/updates/10-schema_compat.update
+++ b/install/updates/10-schema_compat.update
@@ -1,5 +1,7 @@
 dn: cn=sudoers,cn=Schema Compatibility,cn=plugins,cn=config
+only:schema-compat-entry-rdn:'%ifeq("ipaEnabledFlag", "FALSE", "DISABLED", "cn=%{cn}")'
 replace: schema-compat-entry-attribute:'sudoRunAsGroup=%deref("ipaSudoRunAs","cn")::sudoRunAsGroup=%deref_f("ipaSudoRunAsGroup","(objectclass=posixGroup)","cn")'
+
 # Change padding for host and userCategory so the pad returns the same value
 # as the original, '' or -.
 dn: cn=ng,cn=Schema Compatibility,cn=plugins,cn=config
-- 
1.9.3

